dev-guide.md

```diff
@@ -0,0 +1,290 @@
+
+## Local Checks and Tests: Run with One Command
+
+If you are a first-time contributor to EthStorage's Optimism repository or are not yet using `mise` for managing software dependencies, please complete [the steps to set up your development environment](#environment-setup).
+
+In the root directory of your Optimism repo, execute the following command, replacing the placeholders with your [prepared](#prepare-rpc-endpoints) RPC endpoints:
+
+```bash
+# Set your own RPC URLs below. 
+export SEPOLIA_RPC_URL=<YOUR_SEPOLIA_RPC_URL>
+export MAINNET_RPC_URL=<YOUR_MAINNET_RPC_URL>
+mise run dt
+```
+
+## Local Checks and Tests: Step-by-Step Instructions
+
+This section provides a step-by-step guide based on how programming languages are organized in the repository: [Solidity](#solidity) and [Go](#go). 
+
+### Environment Check
+
+Before proceeding, check if `mise` is activated in your current terminal session:
+
+```bash
+mise doctor | grep 'activated:'
+```
+
+The expected output should be:
+
+```bash
+bashactivated: yes
+```
+
+If the output is `no`, ensure that [this step](#activate-mise) has been executed correctly. In some IDE scenarios, you may need to manually activate `mise` each time you open the terminal based on your shell type:
+
+```bash
+eval "$(mise activate zsh)" 
+# or
+eval "$(mise activate bash)"
+```
+
+### Solidity
+
+Navigate to the `contracts-bedrock` directory:
+
+```bash
+cd packages/contracts-bedrock
+```
+
+#### Run Static Checks
+
+Run the following command to clean, build, lint, and check all contracts:
+
+```bash
+just pre-pr
+```
+
+For detailed explanations and potential fixes for specific errors, refer to [this section](#contract-checks-and-fixes-in-detail).
+
+#### Run Unit Tests
+
+Execute the following command to run contract tests using Forge:
+
+```bash
+just test
+```
+
+Review the results and make necessary fixes if errors occur. A successful output may look like this:
+
+```bash
+Ran 282 test suites in 396.48s (1890.89s CPU time): 1763 tests passed, 0 failed, 1 skipped (1764 total tests)
+```
+
+### Go
+
+Navigate to the root directory of your Optimism Monorepo.
+
+#### Lint
+
+Before proceeding with tests, ensure that your Go code is correctly linted:
+
+```bash
+make lint-go
+```
+
+If any lint errors are detected, attempt to fix them with:
+
+```bash
+make lint-go-fix
+```
+Note that some errors may require manual intervention.
+
+#### Build
+
+Go tests require building Go components along with contracts. Execute:
+
+```bash
+# Builds op-node, op-proposer, op-batcher, and contracts-bedrock
+make build
+
+# Build additional essential components
+cd op-program && make op-program-client && cd ..
+cd cannon && make elf && cd ..
+cd op-e2e && make pre-test && cd ..
+```
+
+#### Generate Allocations
+
+Allocations are also required by the following tests. To generate allocations, use:
+
+```bash
+make devnet-allocs
```

Comment:
It's worth mentioning the path to run the command, I guess you're referring to the one at the repo root. Just a heads up that the `make devnet-allocs` are the repo root is gone in the latest develop branch, so this instruction will be stale pretty soon.

Reply:
Good to know. We'll keep the doc and script updated.